### PR TITLE
add parent prop into api

### PR DESCRIPTION
**Pull request by TomasTomecek**
and implement it for github and pagure

### CODE REVIEW

File: ogr/services/github.py [hunk 1]
```diff
@@ -48,6 +48,15 @@ def is_forked(self) -> bool:
     def is_fork(self) -> bool:
         return self.github_repo.fork
 
+    @property
+    def parent(self) -> Optional["GitProject"]:
```
Comment (jpopelka) [inline ogr/services/github.py]:
```suggestion
    def parent(self) -> Optional["GithubProject"]:
```

File: ogr/services/pagure.py [hunk 2]
```diff
@@ -180,6 +180,26 @@ def exists(self):
     def is_fork(self) -> bool:
         return "fork" in self.namespace
 
+    @property
+    def parent(self) -> Optional["GitProject"]:
```
Comment (jpopelka) [inline ogr/services/pagure.py]:
```suggestion
    def parent(self) -> Optional["PagureProject"]:
```

File: ogr/services/pagure.py [hunk 3]
```diff
@@ -180,6 +180,26 @@ def exists(self):
...
+            parent = self._pagure.get_parent()
+            kwargs = self._pagure_kwargs.copy()
+            kwargs.update(
+                repo=self.repo,
+                namespace=parent["namespace"],
+                instance_url=self.instance_url,
+                token=self._token,
+            )
+            if "username" not in kwargs:
+                kwargs["username"] = self.service.user.get_username()
```
Comment (jpopelka) [inline ogr/services/pagure.py]: FYI: you can squash these two lines into
```python
  kwargs.setdefault("username", self.service.user.get_username())
```

Comment (TomasTomecek) [reply ogr/services/pagure.py]: Not my code :P I just copied it.

**Comment by TomasTomecek** [comment]
merge time?

**Pull request merged**